stock_reserve_rule/models/stock_reserve_rule.py

```diff
@@ -133,6 +133,7 @@ class StockReserveRuleRemoval(models.Model):
             ("default", "Default Removal Strategy"),
             ("empty_bin", "Empty Bins"),
             ("packaging", "Full Packaging"),
+            ("by_lot", "By lot"),
```

Comment:
The strategy name is not explicit

Reply:
@jbaudoux any suggestions ?

CC @francesco-ooops

Reply:
@jbaudoux lots are unique by definition :) what about single_lot ?

Reply:
@jbaudoux @francesco-ooops we can follow the example of the packaging (Full Packaging), what do you think about “Full Lot”?

Reply:
@elvise it can be misleading since this IMP also allows to reserve a lot with qty > requested qty, so the "full lot" would not be reserved. I think "Single lot" is the most fitting definition

Reply:
> @elvise it can be misleading since this IMP also allows to reserve a lot with qty > requested qty, so the "full lot" would not be reserved. I think "Single lot" is the most fitting definition

ok let’s go 
@jbaudoux do you agree with Francesco?

Reply:
Yeah, single lot sounds perfect

Reply:
@jbaudoux can we mark this as solved?

Reply:
I'm coming back to this naming as I know understand better what you achieved in your code.
For me, single lot means that you don't want to end up with 2 lots in the reservation but you could take any lot that only partially fill the demand.
However, here you want to reserve a lot that has the required quantity within the tolerance. That's similar to what we already have with "full packaging" or "full bin". So we need to rename to "full lot".